src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -1758,7 +1757,7 @@ void updateZoneSizingEndDay(DataSizing::ZoneSizingData &zsCalcSizing,
     // save heat peak conditions when there is no design heating load or design heating volume flow rate, i.e., when
     // zone temperature is always greater than the zone heating thermostat temperature
     if (zsCalcFinalSizing.DesHeatLoad == 0) {
-        bool FirstIteration = true;
+        bool FirstIteration = true; // declare as static to save for next iteration? but needs to be space/zone specific?
```

Comment:
Move to struct to get expected result? It seems the intent is to set these sizing data on a first pass but is not working as intended.

[src/EnergyPlus/ZoneEquipmentManager.cc:1791]:(style),[unreadVariable],Variable 'FirstIteration' is assigned a value that is never used.

Reply:
If I'm following correctly, this is being used locally here to initialize the peak data to the first timestep values, then let the < (or >) comparisons take over for the remaining timesteps in the loop. So, this really is very local, and it is used in the IF, so not sure why cppCheck thinks it isn't used. The name is misleading, could be `firstTimeStep` or just delete it and put `|| TimeStepIndex = 1` in the `if`.

Reply:
Looking at this again. This is in EndDay calcs so it seems each of these blocks should be executed at least once. That does not appear to be happening for the later blocks where a previous block was executed and the later block doesn't set `bool FirstIteration = true;`. I think the `|| TimeStepIndex == 1` is the better approach. I expect diffs so will target this section in a new branch. I have seen hour of peak = 24 (usually a heating design day) in the past so wondering if TimeStepIndex ==1 or numTimeStepInDay (with condition to not overwrite if set on TimeStepIndex < numTimeStepInDay) is better? If there is no load or flow the entire day maybe it doesn't matter so will try with TimeStepIndex ==1 to see what happens.

Reply:
Oh, and CppCheck thinks `FirstIteration = false;` at line 1791 is never used.